swagger.yaml

```diff
@@ -0,0 +1,367 @@
+      parameters:
+      - $ref: '#/parameters/id'
+      - $ref: '#/parameters/edition'
+      - $ref: '#/parameters/version'
+      - $ref: '#/parameters/filter'
+      responses:
+        200:
+          description: "Json object containing the jobId"
+          schema:
+            $ref: '#/definitions/Job'
+        400:
+          description: "Body of the post request is invalid"
+        404:
+          description: "No dataset edition with a version was found using the id, edition and version provided"
+        500:
+          description: "Failed to process the request due to a internal error"
+definitions:
+  Dataset:
+    type: object
+    description: "An object containing information about published datasets from the ONS"
+    properties:
+      id:
+        type: integer
+        description: "The dataset id"
+      title:
+        type: string
+        description: "The title for the dataset"
+      url:
+        type: string
+        description: "An url to the dataset"
+      releaseDate:
+        type: string
+        description: "The release date of this dataset"
+      edition:
+        type: string
+        description: "The edition of the dataset"
+      version:
+        type: string
+        description: "The version of the dataset"
+      contact:
+        type: object
+        description: "Object containing contact information for this dataset"
+        properties:
+          name:
+            type: string
+            description: "The name of the person"
+          telephone:
+            type: string
+            description: "Telephone number to contact the person"
+          email:
+            type: string
+            description: "An email address to contact the person"
+  Datasets:
+    type: object
+    properties:
+      items:
+        type: array
+        description: "An array of Datasets"
+        items:
+          $ref: '#/definitions/Dataset'
+      numberOfPages:
+        type: integer
+        description: "The number of pages available"
+  Dimension:
+    type: object
+    description: "A single dimension within a dataset"
+    properties:
+      name:
+        type: string
+        description: "The name of the dimension"
+      type:
+        type: string
+        description: "The type of dimension"
+  Dimensions:
+    type: object
+    properties:
+      items:
+        type: array
+        description: "An array of dimensions"
+        items:
+          $ref: '#/definitions/Dimension'
+      numberOfPages:
+        type: integer
+        description: "The number of pages available"
+  DimensionValues:
+    type: object
+    properties:
+      name:
+        type: string
+        description: "The name of the dimension"
+      type:
+        type: string
+        description: "The type of dimension"
+      values:
```

Comment:
How would we imagine this looking with geography? We would need the name of a geographical locations and there corresponding level (town/ward/region etc...)